scripts/link-vms-by-sr.py

```diff
@@ -1,41 +1,37 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
 # Populate a directory of symlinks partitioning VMs by SR
 # (c) Anil Madhavapeddy, Citrix Systems Inc, 2008
 
+from __future__ import print_function
+
+import argparse
 import atexit
+import contextlib
+import os
+import sys
+from pathlib import Path
+
 import XenAPI
-import os, sys
-import getopt
 
-def logout():
-    try:
+
+def logout(session):
+    with contextlib.suppress(Exception):
         session.xenapi.session.logout()
-    except:
-        pass
-atexit.register(logout)
-
-def usage():
-    print >> sys.stderr, "%s [-d <directory>]" % sys.argv[0]
-    sys.exit(1)
-   
-def main(argv):
-    session = XenAPI.xapi_local()
-    session.xenapi.login_with_password("", "", "1.0", "xen-api-scripts-linkvmsbysr.py")
 
-    try:
-        opts, args = getopt.getopt(sys.argv[1:], "hd:", [])
-    except getopt.GetoptError, err:
-        print str(err)
-        usage()
 
-    dir = None
-    for o,a in opts:
-        if o == "-d":
-            dir = a
+def get_input_dir():
+    parser = argparse.ArgumentParser()
```

Comment:
ArgumentParser is usually used in some dedicated function for arg parsers. 
So it can be extended if we need some more args. 

But it is good enough for this case.